### PR TITLE
*: use duration for metric time.

### PR DESCRIPTION
**Pull request by siddontang**
Most of time, we will use time::Duration to timing.

@ngaut @BusyJay @disksing

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/util/mod.rs [hunk 1]
```diff
@@ -246,9 +246,16 @@ impl<'a, T: 'a, V: 'a, E> TryInsertWith<'a, V, E> for Entry<'a, T, V> {
     }
 }
 
+/// Convert Duration to milliseconds.
+pub fn duration_to_ms(d: Duration) -> u64 {
+    let nanos = d.subsec_nanos() as u64;
+    (1_000_000_000 * d.as_secs() + nanos) / (1_000_000)
```
Comment (BusyJay) [inline src/util/mod.rs]: Will it overflow?

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/util/mod.rs [hunk 2]
```diff
@@ -249,7 +249,9 @@ impl<'a, T: 'a, V: 'a, E> TryInsertWith<'a, V, E> for Entry<'a, T, V> {
 /// Convert Duration to milliseconds.
 pub fn duration_to_ms(d: Duration) -> u64 {
     let nanos = d.subsec_nanos() as u64;
-    (1_000_000_000 * d.as_secs() + nanos) / (1_000_000)
+    // Most of case, we can't have so large Duration, so here just panic if overflow now.
```
Comment (BusyJay) [inline src/util/mod.rs]: if you just want panic, then `d.as_secs() * 1000` is enough.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**